app/controllers/application_controller.rb

```diff
@@ -1,10 +1,15 @@
 class ApplicationController < ActionController::Base
   include ApplicationHelper
+
   before_action :set_locale
```

Comment:
callback là gì ?

Reply:
kh co callback ma anh ?

Reply:
`set_locale` được gọi là gì ?
tìm hiểu callback function, note vào đây

Reply:
before_action khoong phai call back ma anh ?

Reply:
set_locale duoc goi la filters

Reply:
https://guides.rubyonrails.org/action_controller_overview.html#after-filters-and-around-filters